### ISSUE TITLE
An exception occurred while fetching time-format data in a cursor

### ISSUE THREAD

**Issue by zhy233** [godror/godror#99]
When I debug, I find that the value of **c.params.Timezone** is nil
and the following if judgment is turned off. Where can I assign the value of  **c.params.Timezone**, or whether the if judgment here can be turned on
```
func (c *conn) Timezone() *time.Location {
	tz := c.params.Timezone
	if false && tz == nil {
		return time.Local
	}
	return tz
}
```
The Timezone() method is in **conn. go**

**Comment by tgulacsi** [comment]
Yes, because c.params.Timezone should be non-nil.
Can you try the latest commit on master?
And we should debug why that params.Timezone is nil.

**Comment by zhy233** [comment]
> Yes, because c.params.Timezone should be non-nil.
> Can you try the latest commit on master?
> And we should debug why that params.Timezone is nil.

I tried. I found that every time the stored procedure was called, the statement preprocessing and the statement execution would call the validation of the database connection, in the following order: ValidateConnection () ===>IsValid() ==> closeNotLocking(). 
In the closeNotLocking method, it sets c.params.timezone to nil. However, when the statement is executed, no value is assigned to c.params.timezone.
This results in the Timezone() method picking c.substanams.timezone nil
